lib/jekyll/theme.rb

```diff
@@ -0,0 +1,40 @@
+module Jekyll
+  class Theme
+    extend Forwardable
+    attr_reader :name
+    def_delegator :gemspec, :version, :version
+
+    def initialize(name)
+      @name = name.downcase.strip
+    end
+
+    def root
+      @root ||= gemspec.full_gem_path
+    end
+
+    def includes_path
+      path_for :includes
+    end
+
+    def layouts_path
+      path_for :layouts
+    end
+
+    def sass_path
+      path_for :sass
+    end
+
+    private
+
+    def path_for(folder)
+      path = Jekyll.sanitized_path root, "_#{folder}"
+      path if Dir.exists?(path)
+    end
+
+    def gemspec
+      @gemspec ||= Gem::Specification.find_by_name(name)
```

Comment:
@benbalter What if we enforced a certain [metadata field](http://guides.rubygems.org/specification-reference/#metadata) in the gemspec, like `s.metadata = { "theme" => true }`?